icu4j/main/classes/core/src/com/ibm/icu/impl/units/SingleUnitImpl.java

```diff
@@ -0,0 +1,157 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+
+package com.ibm.icu.impl.units;
+
+import com.ibm.icu.util.MeasureUnit;
+
+public class SingleUnitImpl {
```

Comment:
Needs class level docs, should probably be final too.

Also, why have an "Impl" class if it's not implementing a public interface?

Normally "FooImpl" is expected to be a non-public implementation of "Foo". What's special about this code?